File: LeetCode/p435.py
from typing import List


class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        
        if len(intervals) <= 1:
            return 0

        intervals.sort(key = lambda x: x[0])
        
        res = [intervals[0]]
        count = 0

        for start, end in intervals[1:]:
            last = res[-1][1] 
            if start < last:
                res[-1][1] = min(last, end)
                count += 1
            else:
                res.append([start, end])

        return count


print(Solution().eraseOverlapIntervals([[1,2],[1,2],[1,2]]))

File: LeetCode-Java/p435.java
import java.util.Arrays;

class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        if (intervals.length <= 1) return 0;
        
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        
        int count = 0;
        int[] prev = intervals[0];
        
        for (int i = 1; i < intervals.length; i++) {
            int[] curr = intervals[i];
            
            if (curr[0] < prev[1]) {
                count++;
                prev[1] = Math.min(prev[1], curr[1]);
            } else {
                prev = curr;
            }
        }
        
        return count;
    }
}
